Mathlib/LinearAlgebra/StdBasis.lean

```diff
@@ -297,6 +297,29 @@ end Module
 
 end Pi
 
+namespace Module
+
+variable (ι R M N : Type*) [Fintype ι] [CommSemiring R]
+  [AddCommMonoid M] [AddCommMonoid N] [Module R M] [Module R N]
+
+/-- The natural linear equivalence: `Mⁱ ≃ Hom(Rⁱ, M)` for an `R`-module `M`. -/
+noncomputable def piEquiv : (ι → M) ≃ₗ[R] ((ι → R) →ₗ[R] M) := Basis.constr (Pi.basisFun R ι) R
+
+lemma piEquiv_apply_apply (v : ι → M) (w : ι → R) :
+    piEquiv ι R M v w = ∑ i, w i • v i := by
+  simp only [piEquiv, Basis.constr_apply_fintype, Basis.equivFun_apply]
+  congr
+
+@[simp] lemma range_piEquiv (v : ι → M) :
+    range (piEquiv ι R M v) = span R (range v) := by
+  ext; simp [mem_span_range_iff_exists_fun, piEquiv_apply_apply]
```

Comment:
Isn't this effectively just [`basis.constr_range`](https://leanprover-community.github.io/mathlib_docs/linear_algebra/basis.html#basis.constr_range)?

Reply:
Yes except I cannot use that without doing a rather intricate dance to handle the redundant `Nonempty ι` variable which `Basis.constr_range` unfortunately demands.

Reply:
Quick fix
```
theorem lmapDomain_supported (f : α → α') (s : Set α) :
    (supported M R s).map (lmapDomain M R f) = supported M R (f '' s) := by
  classical
  cases isEmpty_or_nonempty α
  · simp [s.eq_empty_of_isEmpty]
  refine ...
```

Reply:
I've pushed that change

Reply:
Thanks! It remains to replace the proof of `range_piEquiv`.

Reply:
Oh nice, thanks! I have now replaced the proof of `range_piEquiv`.